src/pages/iou/SplitBillDetailsPage.tsx

```diff
@@ -0,0 +1,169 @@
+                <HeaderWithBackButton title={translate('common.details')} />
+                <View style={[styles.containerWithSpaceBetween, styles.pointerEventsBoxNone]}>
+                    {isScanning && (
+                        <MoneyRequestHeaderStatusBar
+                            title={translate('iou.receiptStatusTitle')}
+                            description={translate('iou.receiptStatusText')}
+                            shouldShowBorderBottom
+                        />
+                    )}
+                    {Boolean(participants.length) && (
+                        <MoneyRequestConfirmationList
+                            hasMultipleParticipants
+                            payeePersonalDetails={payeePersonalDetails}
+                            selectedParticipants={participantsExcludingPayee}
+                            iouAmount={splitAmount}
+                            iouCurrencyCode={splitCurrency}
+                            iouComment={splitComment}
+                            iouCreated={splitCreated}
+                            iouMerchant={splitMerchant}
+                            iouCategory={splitCategory}
+                            iouTag={splitTag}
+                            iouIsBillable={splitBillable}
+                            iouType={CONST.IOU.TYPE.SPLIT}
+                            isReadOnly={!isEditingSplitBill}
+                            shouldShowSmartScanFields
+                            receiptPath={transaction?.receipt?.source}
+                            receiptFilename={transaction?.filename}
+                            shouldShowFooter={false}
+                            isEditingSplitBill={isEditingSplitBill}
+                            hasSmartScanFailed={hasSmartScanFailed}
+                            reportID={reportID}
+                            reportActionID={reportAction?.reportActionID}
+                            transaction={isEditingSplitBill ? draftTransaction || transaction : transaction}
+                            onConfirm={onConfirm}
+                            isPolicyExpenseChat={ReportUtils.isPolicyExpenseChat(report)}
+                            policyID={ReportUtils.isPolicyExpenseChat(report) ? report?.policyID : null}
+                        />
+                    )}
+                </View>
+            </FullPageNotFoundView>
+        </ScreenWrapper>
+    );
+}
+
+SplitBillDetailsPage.displayName = 'SplitBillDetailsPage';
+
+const WrappedComponent = withOnyx<SplitBillDetailsPageProps, Pick<SplitBillDetailsPageOnyxProps, 'transaction' | 'draftTransaction'>>({
+    transaction: {
+        key: ({route, reportActions}) => {
+            const reportAction = reportActions?.[route.params.reportActionID] as (ReportActionBase & OriginalMessageIOU) | undefined;
+            return `${ONYXKEYS.COLLECTION.TRANSACTION}${reportAction?.originalMessage.IOUTransactionID ?? 0}`;
```

Comment:
It should be also possible to narrow down the type without assertion, for example. But it's up to you
```suggestion
            const reportAction = reportActions?.[route.params.reportActionID];
            const IOUTransactionID = reportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU && reportAction?.originalMessage?.IOUTransactionID ? reportAction.originalMessage.IOUTransactionID : 0;
            return `${ONYXKEYS.COLLECTION.TRANSACTION}${IOUTransactionID}`;
```

Reply:
@VickyStash Thank you! I like your way better! 😄